cranelift/codegen/src/machinst/blockorder.rs

```diff
@@ -253,6 +253,18 @@ impl BlockLoweringOrder {
             block_in_count[entry] += 1;
         }
 
+        // All blocks ending in conditional branches have at least two
+        // successors as far as edge-splitting is concerned, even if
+        // they only really have one (e.g. br_table with no table,
+        // just a default label).
```

Comment:
I'm not seeing how a `br_table` with no table and just a default label has an outgoing edge count of two.

The explanation you wrote in the PR description makes sense to me, but I'm having trouble matching that with what is written/happening here. I felt like I understood the problem when reading the PR description, but looking at the fix I'm now lost again.

Is it that there is some conservative over-estimating happening here? If so, then I think this comment needs to better describe what is going on. If not, then I think this comment also needs to better describe what is going on :-p

Reply:
Err, this was a way to force the edge-splitting not to assume it can add moves at the end of this block (there aren't actually two successors, but if we say there are, then the logic moves onto next options); but I agree the comment could be clearer! I'll update.